### DISCUSSION TITLE
Prime React DataTable is Laggy and Jittery

### DISCUSSION THREAD

**Discussion by abdulrehmanwaseem** [community/community#140314] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

General

### Body

Is it just me, or is the Prime React DataTable search input really laggy and jittery when filtering data? Whenever I type, there's a noticeable delay between each keystroke and the corresponding filter action, especially when dealing with large datasets. I've tried debouncing, pagination, lazy loading, and even memoization with React.memo and useMemo, but the issue persists.

Is anyone else facing this, or is there a better way to optimize the search functionality in Prime React’s DataTable? Any tips or insights would be super helpful! 🙏

Thanks in advance!

Abdul Rehman

**Comment by mohdsaadshaikh** [comment]
Hey Abdul Rehman,

This happens to me as well! I've been experiencing the same laggy and jittery behavior with the search input in Prime React DataTable, especially with large datasets. I've tried a few different optimizations like debouncing and lazy loading, but nothing seems to fully resolve the issue.

Let’s hope they address this in a future update and improve the performance soon. 🤞

Best,
Saad**

**Comment by abdulrehmanwaseem** [comment]
Got The Answer By mohdsaadshaikh

**Comment by ebndev** [comment, marked as answer]
Hi @abdulrehmanwaseem & @mohdsaadshaikh, thanks for being a part of the GitHub Community!

While we appreciate you wanting to contribute to Discussions, please note: We made the decision to disable the ability to earn Achievements in our Community in order to discourage users from participating in coordinated or inauthentic activity like rapid questions and answers in order to earn badges. You can learn more about this decision in our announcement post here [Achievements will no longer be available in the Community](https://github.com/orgs/community/discussions/106536).

Note that GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this, and other related posts. 

Any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
